scripts/cut-release.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+# Increment a version string using Semantic Versioning (SemVer) terminology.
+
+# Parse command line options.
+
+while getopts ":Mmp" Option
+do
+  case $Option in
+    M ) major=true;;
+    m ) minor=true;;
+  esac
+done
+
+shift $(($OPTIND - 1))
+
+version=$(cat VERSION)
+
+if [[ -z $major ]] && [[ -z $minor ]]; then
+  minor=true
+fi
+
+# Build array from version string.
+
+a=( ${version//./ } )
+
+# If version string is missing or has the wrong number of members, show usage message.
+
+if [ ${#a[@]} -ne 3 ]
+then
+  echo "usage: $(basename $0) [-Mmp]"
+  exit 1
+fi
+
+# Increment version numbers as requested.
+
+if [ ! -z $major ]
+then
+  ((a[0]++))
+  a[1]=0
+  a[2]=0
+fi
+
+if [ ! -z $minor ]
+then
+  ((a[1]++))
+  a[2]=0
+fi
+
+newversion=${a[0]}.${a[1]}.${a[2]}
+releasebranch="${a[0]}.${a[1]}.x"
+echo "Cutting release branch $newversion"
```

Comment:
This `newversion` has two issues:

* Logically `develop` doesn't represent the current release version, but the development version of the next release.
* `release/A.B.x` starts with `A.B.0`, but might be bumped to `A.B.1` and further, but `develop` stays as `A.B.0`.

Although it is true that it would complicate this a bit, since you would need to do two commits.
But I don't think that's that bad, and once it's implemented in the script it should work fine.

Reply:
Since there is not direct gain from this other than logical/syntactical correctness it's not strictly necessary, but I'd expect this or something like this to become necessary in the future. But I'm fine with it as it is right now.